src/main/scala/zio/json/decoder.scala

```diff
@@ -228,6 +230,29 @@ object Decoder extends GeneratedTuples with DecoderLowPriority1 with DecoderLowP
       }
     }
 
+  implicit def vector[A](implicit A: Decoder[A]): Decoder[Vector[A]] =
+    list[A].map(_.toVector)
+
+  implicit def seq[A](implicit A: Decoder[A]): Decoder[Seq[A]] =
+    list[A].map(_.toSeq)
+
+  implicit def set[A](implicit A: Decoder[A]): Decoder[Set[A]] =
+    list[A].map(_.toSet)
+
+  implicit def array[A:ClassTag](implicit A: Decoder[A]): Decoder[Array[A]] =
```

Comment:
I'm don't think it is in ZIO's interests to support mutable data structures in the public API.

Reply:
There is another reason not to support `Array[A]`. Since it is so primitive, it is likely that if somebody wants to support it they will want to support it on a per type basis (in Haskell terms, `OverlappingInstances`). e.g. `Array[Byte]` as base16/58/64, `Array[Char]` as a `String`, etc etc. In which case, it's best left free for the user to define.

Reply:
Removed. Good point.